runtime/gateway.go

```diff
@@ -322,24 +317,23 @@ func (gateway *Gateway) setupMetrics(config *StaticConfig) (err error) {
 		panic("expected gateway to have MetricsBackend in opts")
 	}
 
-	// Adding 'env' and 'service' common tags, since they are not emitted by metrics backend.
+	// TODO: Remove 'env' and 'service' default tags once they are emitted by metrics backend.
 	defaultTags := map[string]string{
 		"env":     env,
 		"service": service,
 	}
 	gateway.RootScope, gateway.scopeCloser = tally.NewRootScope(
 		tally.ScopeOptions{
 			Tags:           defaultTags,
-			Prefix:         service + ".production",
 			CachedReporter: gateway.metricsBackend,
 			Separator:      tally.DefaultSeparator,
 		},
-		time.Duration(config.MustGetInt("metrics.tally.flushInterval"))*time.Millisecond,
+		time.Duration(config.MustGetInt("metrics.flushInterval"))*time.Millisecond,
 	)
-	gateway.MetricsScope = gateway.RootScope.SubScope("all-workers")
-	gateway.PerWorkerMetricsScope = gateway.RootScope.SubScope("per-worker").Tagged(map[string]string{
-		"host": GetHostname(),
-	})
+	// As per M3 best practices, creating separate all-host and per-host metrics
+	// to reduce metric cardinality when querying metrics for all hosts.
+	gateway.AllHostScope = gateway.RootScope.Tagged(map[string]string{"host": "all"})
+	gateway.PerHostScope = gateway.RootScope.Tagged(map[string]string{"host": GetHostname()})
```

Comment:
We want the per host scope to be prefixed with some unique prefix to make grafana querying easier.